community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -1002,13 +1007,10 @@ private StatementOperationParts buildStatementOperations(
         parts = parts.override( null, null, null, lockingContext, lockingContext, lockingContext, lockingContext,
                 lockingContext, null, null, null );
         // + Guard
-        if ( guard != null )
-        {
-            GuardingStatementOperations guardingOperations = new GuardingStatementOperations(
-                    parts.entityWriteOperations(), parts.entityReadOperations(), guard );
-            parts = parts.override( null, null, guardingOperations, guardingOperations, null, null, null, null,
-                    null, null, null );
-        }
+        GuardingStatementOperations guardingOperations = new GuardingStatementOperations(
```

Comment:
Then perhaps we should have a `dbms.guard.enabled` setting as well, since it's unnecessary to have all users that don't use the guard pay for this extra layer.